src/raft.rs

```diff
@@ -884,16 +994,132 @@ impl<T: Storage> Raft<T> {
         self.r.send_append_aggressively(to, pr, &mut self.msgs)
     }
 
+    // Find an appropriate agent.
+    // If found an appropriate one, mark the corresponding message's type as
+    // MsgGroupBroadcast and return true. If not found, return false.
+    fn select_agent_for_bcast_group(&self, msgs: &mut [Message]) -> bool {
```

Comment:
Why not return the index of the message? Changing the type in this function is quite confusing.

Reply:
I think it is a better choice.